topology/src/test/java/io/camunda/zeebe/topology/PersistedClusterTopologyTest.java

```diff
@@ -9,86 +9,77 @@
-import io.camunda.zeebe.topology.state.ClusterChangePlan;
 import io.camunda.zeebe.topology.state.ClusterTopology;
-import io.camunda.zeebe.topology.state.CompletedChange;
-import io.camunda.zeebe.topology.state.MemberState;
-import io.camunda.zeebe.topology.state.TopologyChangeOperation;
-import io.camunda.zeebe.util.ReflectUtil;
 import java.io.IOException;
 import java.nio.file.Files;
-import net.jqwik.api.Arbitraries;
-import net.jqwik.api.Arbitrary;
-import net.jqwik.api.Combinators;
-import net.jqwik.api.ForAll;
-import net.jqwik.api.Property;
-import net.jqwik.api.Provide;
-import net.jqwik.api.domains.Domain;
-import net.jqwik.api.domains.DomainContext;
-import net.jqwik.api.domains.DomainContextBase;
+import java.nio.file.StandardOpenOption;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
 
 final class PersistedClusterTopologyTest {
+  @Test
+  void shouldDetectFileWithMoreDataThanExpected() throws IOException {
+    // given
+    final var tmp = Files.createTempDirectory("topology");
+    final var topologyFile = tmp.resolve("topology.meta");
+    final var serializer = new ProtoBufSerializer();
+    final var persistedClusterTopology = PersistedClusterTopology.ofFile(topologyFile, serializer);
+    persistedClusterTopology.update(ClusterTopology.init());
+
+    // when
+    Files.write(topologyFile, "more data than expected".getBytes(), StandardOpenOption.APPEND);
 
-  @Property(tries = 100)
-  @Domain(ClusterTopologyDomain.class)
-  @Domain(DomainContext.Global.class)
-  void shouldUpdatePersistedFile(
-      @ForAll final ClusterTopology initialTopology, @ForAll final ClusterTopology updatedTopology)
-      throws IOException {
+    // then
+    Assertions.assertThatCode(() -> PersistedClusterTopology.ofFile(topologyFile, serializer))
+        .hasMessageContaining("checksum");
+  }
+
+  @Test
+  void shouldDetectFileWithBrokenHeader() throws IOException {
     // given
     final var tmp = Files.createTempDirectory("topology");
     final var topologyFile = tmp.resolve("topology.meta");
     final var serializer = new ProtoBufSerializer();
-    final var persistedClusterTopology = new PersistedClusterTopology(topologyFile, serializer);
+    final var persistedClusterTopology = PersistedClusterTopology.ofFile(topologyFile, serializer);
+    persistedClusterTopology.update(ClusterTopology.init());
 
     // when
-    persistedClusterTopology.update(initialTopology);
-    persistedClusterTopology.update(updatedTopology);
+    Files.write(topologyFile, "broken header".getBytes(), StandardOpenOption.WRITE);
 
     // then
-    assertEquals(updatedTopology, persistedClusterTopology.getTopology());
-    assertEquals(
-        updatedTopology, serializer.decodeClusterTopology(Files.readAllBytes(topologyFile)));
+    Assertions.assertThatCode(() -> PersistedClusterTopology.ofFile(topologyFile, serializer))
+        .hasMessageContaining("version");
```

Comment:
:wrench: imo would be better to expose the exception type and assert that than messages, it's more explicit _and_ it works well with automated refactoring